### PR TITLE
refactor(core): Remove attohttpc in favor of reqwest

### PR DESCRIPTION
**Pull request by FabianLars**
ref: for example https://github.com/tauri-apps/tauri/issues/6415 and https://github.com/tauri-apps/tauri/issues/5344

### What kind of change does this PR introduce?

- [x] Bugfix - yes i count this as a bugfix considering how many issues users had with atto
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No, well... Technically it is because of the enum variants but i think it's worth it.

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

**Comment by FabianLars** [comment]
that clippy json output gives me a headache lol. Also, no idea why i couldn't trigger that warning locally 🤔

**Comment by lucasfernog** [comment]
Usually that happens when I don't have latest cargo. Or not using the right feature flags.

File: core/tauri/src/api/error.rs [hunk 1]
```diff
@@ -22,20 +22,13 @@ pub enum Error {
   #[error("user cancelled the dialog")]
   DialogCancelled,
   /// The network error.
-  #[cfg(all(feature = "http-api", not(feature = "reqwest-client")))]
-  #[error("Network Error: {0}")]
-  Network(#[from] attohttpc::Error),
-  /// The network error.
-  #[cfg(feature = "reqwest-client")]
-  #[cfg_attr(doc_cfg, doc(cfg(feature = "reqwest-client")))]
+  #[cfg(all(feature = "http-api"))]
```
Comment (lucasfernog) [inline core/tauri/src/api/error.rs]:
```suggestion
  #[cfg(feature = "http-api")]
```

**Comment by FabianLars** [comment]
ahhhhh, i was in the root dir and not in core/tauri ...

File: .changes/remove-attohttpc.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+'tauri': patch
+'tauri-bundler': patch
+---
+
+Removed `attohttpc` in favor of `reqwest` (core) and `ureq` (tooling).
```
Comment (lucasfernog) [inline .changes/remove-attohttpc.md]: I prefer a separate change file for each package in this case.

**Review by lucasfernog — CHANGES_REQUESTED** [review]

File: .changes/remove-attohttpc.md (hunk 2 above)
Comment (FabianLars) [reply .changes/remove-attohttpc.md]: right, makes sense! Still trying to get a feeling for changefiles

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**